### PR TITLE
Add support for CredentialProvider 

### PR DESCRIPTION
**Pull request by vaikas**
This allows the customer to create a credential that should be used when talking to github (for example private repos). The client is not plumbed through all the way since there's some merging going on, but at least it's plumbed through on the backend.

### CODE REVIEW

**Comment by vaikas-google** [comment]
@bmelville

File: manager/deployments.go [hunk 1]
```diff
@@ -433,3 +447,79 @@ func getDownloadURLsHandlerFunc(w http.ResponseWriter, r *http.Request) {
 
 	util.LogHandlerExitWithJSON(handler, w, c, http.StatusOK)
 }
+
+func getCredential(w http.ResponseWriter, r *http.Request, handler string) *common.RegistryCredential {
+	util.LogHandlerEntry(handler, r)
+	b := io.LimitReader(r.Body, *maxLength*1024)
```
Comment (bmelville) [inline manager/deployments.go]: This whole bit of code is duplicated all over the place, can we split this out to a function like getYamlFromRequest

File: registry/registryprovider.go [hunk 2]
```diff
@@ -133,22 +137,63 @@ type GithubRegistryProvider interface {
...
+	if c != nil {
+		if c.APIToken != "" {
+			ts := oauth2.StaticTokenSource(
+				&oauth2.Token{AccessToken: string(c.APIToken)},
+			)
+			tc := oauth2.NewClient(oauth2.NoContext, ts)
+			return github.NewClient(tc), nil
+		}
+		if c.BasicAuth.Username != "" && c.BasicAuth.Password != "" {
+
```
Comment (bmelville) [inline registry/registryprovider.go]: The suspense is killing me

File: dm/dm.go [hunk 3]
```diff
@@ -122,10 +122,11 @@ func getRegistryProvider() registry.RegistryProvider {
...
 	return &common.Registry{
-		Name:   util.TrimURLScheme(URL),
-		Type:   common.GithubRegistryType,
-		URL:    URL,
-		Format: common.RegistryFormat(tFormat),
+		Name:           util.TrimURLScheme(URL),
+		Type:           common.GithubRegistryType,
+		URL:            URL,
+		Format:         common.RegistryFormat(tFormat),
+		CredentialName: "default",
```
Comment (bmelville) [inline dm/dm.go]: Do we set a default credential somewhere? Should this just be no credential? Might be confusing to make them specify a credential when getting started (or did I miss something?)

Comment (vaikas-google) [inline dm/dm.go]: Discussed offline. But if a credential is not found, we fall back to unauthenticated, hence the behaviour is like it was before this CL.

File: manager/deployments.go (hunk 1 above)
Comment (vaikas-google) [inline manager/deployments.go]: Done, but since it converts to json, called it getJsonFromRequest

File: registry/registryprovider.go (hunk 2 above)
Comment (vaikas-google) [inline registry/registryprovider.go]: :) Fixed.

**Comment by vaikas-google** [comment]
Thanks for the review. PTAL.

**Pull request merged**